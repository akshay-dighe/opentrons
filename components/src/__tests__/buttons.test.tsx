// button component tests
import React from 'react'
import Renderer from 'react-test-renderer'

import {
  Button,
  FlatButton,
  DeprecatedPrimaryButton,
  OutlineButton,
  IconButton,
} from '..'

describe('buttons', () => {
  const onClick = jest.fn()

  afterEach(() => {
    jest.resetAllMocks()
  })

  it('creates a button with props', () => {
    const button = Renderer.create(
      <Button
        onClick={onClick}
        className="class"
        title="title"
        disabled={false}
      >
        children
      </Button>
    ).root.findByType('button')

    button.props.onClick()
    expect(button.props.className).toMatch(/\bclass\b/)
    expect(button.props.title).toBe('title')
    expect(button.props.disabled).toBe(false)
    expect(onClick).toHaveBeenCalled()
  })

  it('disabled sets onClick to undefined', () => {
    const button = Renderer.create(
      <Button onClick={onClick} disabled />
    ).root.findByType('button')

    expect(button.props.disabled).toBe(true)
    expect(button.props.onClick).toBe(undefined)
  })

  it('Button renders correctly', () => {
    const tree = Renderer.create(
      <Button onClick={onClick} title="t" className="c">
        children
      </Button>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Button with iconName renders correctly', () => {
    const tree = Renderer.create(
      <Button
        onClick={onClick}
        title="t"
        className="c"
        iconName="flask-outline"
      >
        children
      </Button>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('DeprecatedPrimaryButton renders correctly', () => {
    const tree = Renderer.create(
      <DeprecatedPrimaryButton onClick={onClick} title="t" className="c">
        children
      </DeprecatedPrimaryButton>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('DeprecatedPrimaryButton with iconName renders correctly', () => {
    const tree = Renderer.create(
      <DeprecatedPrimaryButton
        onClick={onClick}
        title="t"
        className="c"
        iconName="flask-outline"
      >
        children
      </DeprecatedPrimaryButton>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('FlatButton renders correctly', () => {
    const tree = Renderer.create(
      <FlatButton onClick={onClick} title="t" className="c">
        children
      </FlatButton>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('FlatButton with iconName renders correctly', () => {
    const tree = Renderer.create(
      <FlatButton
        onClick={onClick}
        title="t"
        className="c"
        iconName="flask-outline"
      >
        children
      </FlatButton>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('OutlineButton renders correctly', () => {
    const tree = Renderer.create(
      <OutlineButton onClick={onClick} title="t" className="c">
        children
      </OutlineButton>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Inverted OutlineButton renders correctly', () => {
    const tree = Renderer.create(
      <OutlineButton onClick={onClick} title="t" className="c" inverted>
        children
      </OutlineButton>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('OutlineButton with iconName renders correctly', () => {
    const tree = Renderer.create(
      <OutlineButton
        onClick={onClick}
        title="t"
        className="c"
        iconName="flask-outline"
      >
        children
      </OutlineButton>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('IconButton renders correctly', () => {
    const tree = Renderer.create(
      <IconButton
        onClick={onClick}
        title="t"
        className="c"
        name="close"
        spin
        disabled
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
