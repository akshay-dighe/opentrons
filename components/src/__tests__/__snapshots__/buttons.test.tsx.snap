// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buttons Button renders correctly 1`] = `
<button
  className="c"
  onClick={[MockFunction]}
  title="t"
  type="button"
>
  children
</button>
`;

exports[`buttons Button with iconName renders correctly 1`] = `
.c0.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<button
  className="c"
  onClick={[MockFunction]}
  title="t"
  type="button"
>
  <svg
    aria-hidden="true"
    className=" c0 icon"
    fill="currentColor"
    version="1.1"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2.44444444,20.4 C2.44444444,21.0627417 2.99165197,21.6 3.66666667,21.6 L18.3333333,21.6 C19.008348,21.6 19.5555556,21.0627417 19.5555556,20.4 C19.5555556,20.148 19.47,19.908 19.3355556,19.716 L12.2222222,7.62 L12.2222222,2.4 L9.77777778,2.4 L9.77777778,7.62 L2.66444444,19.716 C2.53,19.908 2.44444444,20.148 2.44444444,20.4 Z M3.66666667,24 C1.64162258,24 0,22.3882251 0,20.4 C0,19.68 0.22,19.008 0.611111111,18.444 L7.33333333,6.972 L7.33333333,4.8 C6.65831864,4.8 6.11111111,4.2627417 6.11111111,3.6 L6.11111111,2.4 C6.11111111,1.0745166 7.20552617,0 8.55555556,0 L13.4444444,0 C14.7944738,0 15.8888889,1.0745166 15.8888889,2.4 L15.8888889,3.6 C15.8888889,4.2627417 15.3416814,4.8 14.6666667,4.8 L14.6666667,6.972 L21.3888889,18.444 C21.78,19.008 22,19.68 22,20.4 C22,22.3882251 20.3583774,24 18.3333333,24 L3.66666667,24 Z M12.2222222,16.8 L13.86,15.192 L16.2188889,19.2 L5.78111111,19.2 L9.03222222,13.668 L12.2222222,16.8 Z M11.6111111,12 C11.9486185,12 12.2222222,12.2686292 12.2222222,12.6 C12.2222222,12.9313708 11.9486185,13.2 11.6111111,13.2 C11.2736038,13.2 11,12.9313708 11,12.6 C11,12.2686292 11.2736038,12 11.6111111,12 Z"
      fillRule="evenodd"
    />
  </svg>
  children
</button>
`;

exports[`buttons DeprecatedPrimaryButton renders correctly 1`] = `
<button
  className="button_primary c"
  onClick={[MockFunction]}
  title="t"
  type="button"
>
  children
</button>
`;

exports[`buttons DeprecatedPrimaryButton with iconName renders correctly 1`] = `
.c0.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<button
  className="button_primary c"
  onClick={[MockFunction]}
  title="t"
  type="button"
>
  <svg
    aria-hidden="true"
    className=" c0 icon"
    fill="currentColor"
    version="1.1"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2.44444444,20.4 C2.44444444,21.0627417 2.99165197,21.6 3.66666667,21.6 L18.3333333,21.6 C19.008348,21.6 19.5555556,21.0627417 19.5555556,20.4 C19.5555556,20.148 19.47,19.908 19.3355556,19.716 L12.2222222,7.62 L12.2222222,2.4 L9.77777778,2.4 L9.77777778,7.62 L2.66444444,19.716 C2.53,19.908 2.44444444,20.148 2.44444444,20.4 Z M3.66666667,24 C1.64162258,24 0,22.3882251 0,20.4 C0,19.68 0.22,19.008 0.611111111,18.444 L7.33333333,6.972 L7.33333333,4.8 C6.65831864,4.8 6.11111111,4.2627417 6.11111111,3.6 L6.11111111,2.4 C6.11111111,1.0745166 7.20552617,0 8.55555556,0 L13.4444444,0 C14.7944738,0 15.8888889,1.0745166 15.8888889,2.4 L15.8888889,3.6 C15.8888889,4.2627417 15.3416814,4.8 14.6666667,4.8 L14.6666667,6.972 L21.3888889,18.444 C21.78,19.008 22,19.68 22,20.4 C22,22.3882251 20.3583774,24 18.3333333,24 L3.66666667,24 Z M12.2222222,16.8 L13.86,15.192 L16.2188889,19.2 L5.78111111,19.2 L9.03222222,13.668 L12.2222222,16.8 Z M11.6111111,12 C11.9486185,12 12.2222222,12.2686292 12.2222222,12.6 C12.2222222,12.9313708 11.9486185,13.2 11.6111111,13.2 C11.2736038,13.2 11,12.9313708 11,12.6 C11,12.2686292 11.2736038,12 11.6111111,12 Z"
      fillRule="evenodd"
    />
  </svg>
  children
</button>
`;

exports[`buttons FlatButton renders correctly 1`] = `
<button
  className="button_flat c"
  onClick={[MockFunction]}
  title="t"
  type="button"
>
  children
</button>
`;

exports[`buttons FlatButton with iconName renders correctly 1`] = `
.c0.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<button
  className="button_flat c"
  onClick={[MockFunction]}
  title="t"
  type="button"
>
  <svg
    aria-hidden="true"
    className=" c0 icon"
    fill="currentColor"
    version="1.1"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2.44444444,20.4 C2.44444444,21.0627417 2.99165197,21.6 3.66666667,21.6 L18.3333333,21.6 C19.008348,21.6 19.5555556,21.0627417 19.5555556,20.4 C19.5555556,20.148 19.47,19.908 19.3355556,19.716 L12.2222222,7.62 L12.2222222,2.4 L9.77777778,2.4 L9.77777778,7.62 L2.66444444,19.716 C2.53,19.908 2.44444444,20.148 2.44444444,20.4 Z M3.66666667,24 C1.64162258,24 0,22.3882251 0,20.4 C0,19.68 0.22,19.008 0.611111111,18.444 L7.33333333,6.972 L7.33333333,4.8 C6.65831864,4.8 6.11111111,4.2627417 6.11111111,3.6 L6.11111111,2.4 C6.11111111,1.0745166 7.20552617,0 8.55555556,0 L13.4444444,0 C14.7944738,0 15.8888889,1.0745166 15.8888889,2.4 L15.8888889,3.6 C15.8888889,4.2627417 15.3416814,4.8 14.6666667,4.8 L14.6666667,6.972 L21.3888889,18.444 C21.78,19.008 22,19.68 22,20.4 C22,22.3882251 20.3583774,24 18.3333333,24 L3.66666667,24 Z M12.2222222,16.8 L13.86,15.192 L16.2188889,19.2 L5.78111111,19.2 L9.03222222,13.668 L12.2222222,16.8 Z M11.6111111,12 C11.9486185,12 12.2222222,12.2686292 12.2222222,12.6 C12.2222222,12.9313708 11.9486185,13.2 11.6111111,13.2 C11.2736038,13.2 11,12.9313708 11,12.6 C11,12.2686292 11.2736038,12 11.6111111,12 Z"
      fillRule="evenodd"
    />
  </svg>
  children
</button>
`;

exports[`buttons IconButton renders correctly 1`] = `
.c0.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<button
  className="button_flat button_icon c"
  disabled={true}
  title="t"
  type="button"
>
  <svg
    aria-hidden="true"
    className=" c0 spin"
    fill="currentColor"
    version="1.1"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18 7.209L16.791 6 12 10.791 7.209 6 6 7.209 10.791 12 6 16.791 7.209 18 12 13.209 16.791 18 18 16.791 13.209 12z"
      fillRule="evenodd"
    />
  </svg>
</button>
`;

exports[`buttons Inverted OutlineButton renders correctly 1`] = `
<button
  className="button_outline c inverted"
  onClick={[MockFunction]}
  title="t"
  type="button"
>
  children
</button>
`;

exports[`buttons OutlineButton renders correctly 1`] = `
<button
  className="button_outline c"
  onClick={[MockFunction]}
  title="t"
  type="button"
>
  children
</button>
`;

exports[`buttons OutlineButton with iconName renders correctly 1`] = `
.c0.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<button
  className="button_outline c"
  onClick={[MockFunction]}
  title="t"
  type="button"
>
  <svg
    aria-hidden="true"
    className=" c0 icon"
    fill="currentColor"
    version="1.1"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2.44444444,20.4 C2.44444444,21.0627417 2.99165197,21.6 3.66666667,21.6 L18.3333333,21.6 C19.008348,21.6 19.5555556,21.0627417 19.5555556,20.4 C19.5555556,20.148 19.47,19.908 19.3355556,19.716 L12.2222222,7.62 L12.2222222,2.4 L9.77777778,2.4 L9.77777778,7.62 L2.66444444,19.716 C2.53,19.908 2.44444444,20.148 2.44444444,20.4 Z M3.66666667,24 C1.64162258,24 0,22.3882251 0,20.4 C0,19.68 0.22,19.008 0.611111111,18.444 L7.33333333,6.972 L7.33333333,4.8 C6.65831864,4.8 6.11111111,4.2627417 6.11111111,3.6 L6.11111111,2.4 C6.11111111,1.0745166 7.20552617,0 8.55555556,0 L13.4444444,0 C14.7944738,0 15.8888889,1.0745166 15.8888889,2.4 L15.8888889,3.6 C15.8888889,4.2627417 15.3416814,4.8 14.6666667,4.8 L14.6666667,6.972 L21.3888889,18.444 C21.78,19.008 22,19.68 22,20.4 C22,22.3882251 20.3583774,24 18.3333333,24 L3.66666667,24 Z M12.2222222,16.8 L13.86,15.192 L16.2188889,19.2 L5.78111111,19.2 L9.03222222,13.668 L12.2222222,16.8 Z M11.6111111,12 C11.9486185,12 12.2222222,12.2686292 12.2222222,12.6 C12.2222222,12.9313708 11.9486185,13.2 11.6111111,13.2 C11.2736038,13.2 11,12.9313708 11,12.6 C11,12.2686292 11.2736038,12 11.6111111,12 Z"
      fillRule="evenodd"
    />
  </svg>
  children
</button>
`;
